### ISSUE TITLE
Arcane variable handling

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#484]
```
# Scryer
$ scryer-prolog -f
?- use_module(library(lambda)).
   true.
?- A = 1, Z = [Z,A|Z].
   A = 1, Z = [Z,1|Z]. % expected
?- call((A,Z)+\Z^A^(Z=[Z,A|Z]),Z,1).
   A = 1, Z = [Z,1|Z]. % expected
?- call((A,Z)+\Z^A^(Z=[A,Z|Z]),Z,1).
   A = 1, Z = [1,Z|Z]. % expected

# Trealla
$ tpl
?- use_module(library(lambda)).
   true.
?- A = 1, Z = [Z,A|Z].
   A = 1, Z = [Z,1|A]. % unexpected
?- call((A,Z)+\Z^A^(Z=[Z,A|Z]),Z,1).
   A = 1, Z = [_,1|Z]. % unexpected
?- call((A,Z)+\Z^A^(Z=[A,Z|Z]),Z,1).
   A = 1, Z = [1,_|_A]. % unexpected
```

**Comment by flexoron** [comment]
```
$ scryer-prolog
?- use_module(library(lists)).
   true.
?- use_module(library(lambda)).
   true.
?- call((C)+\A^B^C^append(A,B,C),[Z],[Z|Z],[X,Z,Z,Z|Z]).
   C = [[Z,Z|Z],[Z,Z|Z],Z,Z|Z],
   Z = [Z,Z|Z],
   X = [Z,Z|Z].

$ tpl
?- use_module(library(lambda)).
   true.
?- call((C)+\A^B^C^append(A,B,C),[Z],[Z|Z],[X,Z,Z,Z|Z]).
   C = [[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C],[C,C|C]|...],
   Z = [A,Z|Z],
   X = [B,X|X].
```

**Comment by flexoron** [comment]
Fixed with v2.34.28
